tests/unittests/tests-hashes/tests-hashes-sha256-hmac.c

```diff
@@ -0,0 +1,172 @@
+    key[0] = 'k';
+    key[1] = 'e';
+    key[2] = 'y';
+
+    hmac_sha256(key, 3, (unsigned*)str, strlen(str), hmac);
+    TEST_ASSERT(compare_str_vs_digest(
+                 "f7bc83f430538424b13298e6aa6fb143ef4d59a14946175997479dbc2d1a3cd8", hmac));
+}
+
+/*
+        The followig testcases are taken from:
+        https://tools.ietf.org/html/rfc4868#section-2.7.1
+*/
+
+static void test_hashes_hmac_sha256_hash_PRF1(void)
+{
+    /* Test Case PRF-1: */
+    const char strPRF1[] = "Hi There";
+    unsigned char key[20];
+    static unsigned char hmac[SHA256_DIGEST_LENGTH];
+    memset(key, 0x0b, 20);
+
+    hmac_sha256(key, 20, (unsigned*)strPRF1, strlen(strPRF1), hmac);
+    TEST_ASSERT(compare_str_vs_digest(
+                 "b0344c61d8db38535ca8afceaf0bf12b881dc200c9833da726e9376c2e32cff7", hmac));
+}
+
+static void test_hashes_hmac_sha256_hash_PRF2(void)
+{
+    /* Test Case PRF-2: */
+    const char strPRF2[] = "what do ya want for nothing?";
+    unsigned char key[4] = {'J', 'e', 'f', 'e'};
+    static unsigned char hmac[SHA256_DIGEST_LENGTH];
+
+    hmac_sha256(key, 4, (unsigned*)strPRF2, strlen(strPRF2), hmac);
+    TEST_ASSERT(compare_str_vs_digest(
+                 "5bdcc146bf60754e6a042426089575c75a003f089d2739839dec58b964ec3843", hmac));
+}
+
+static void test_hashes_hmac_sha256_hash_PRF3(void)
+{
+    /* Test Case PRF-3: */
+    char strPRF3[50];
+    unsigned char key[20];
+    static unsigned char hmac[SHA256_DIGEST_LENGTH];
+
+    memset(strPRF3, 0xdd, 50);
+    memset(key, 0xaa, 20);
+
+    hmac_sha256(key, 20, (unsigned*)strPRF3, 50, hmac);
+    TEST_ASSERT(compare_str_vs_digest(
+                 "773ea91e36800e46854db8ebd09181a72959098b3ef8c122d9635514ced565fe", hmac));
+}
+
+static void test_hashes_hmac_sha256_hash_PRF4(void)
+{
+    /* Test Case PRF-4: */
+    char strPRF4[50];
+    unsigned char key[25];
+    static unsigned char hmac[SHA256_DIGEST_LENGTH];
+
+    memset(strPRF4, 0xcd, 50);
+    /*
+    * set key to: 0102030405060708090a0b0c0d0e0f10111213141516171819
+    */
+    for (size_t i = 0; i < 25; ++i) {
+        key[i] = i+1;
+    }
+
+    hmac_sha256(key, 25, (unsigned*)strPRF4, 50, hmac);
+    TEST_ASSERT(compare_str_vs_digest(
+                 "82558a389a443c0ea4cc819899f2083a85f0faa3e578f8077a2e3ff46729665b", hmac));
+}
+
+static void test_hashes_hmac_sha256_hash_PRF5(void)
+{
+    /* Test Case PRF-5: */
+    const char strPRF5[] = "Test Using Larger Than Block-Size Key - Hash Key First";
+    unsigned char longKey[131];
+    static unsigned char hmac[SHA256_DIGEST_LENGTH];
+    memset(longKey, 0xaa, 131);
+
+    hmac_sha256(longKey, 131, (unsigned*)strPRF5, strlen(strPRF5), hmac);
+    TEST_ASSERT(compare_str_vs_digest(
+                 "60e431591ee0b67f0d8a26aacbf5b77f8e0bc6213728c5140546040f0ee37f54", hmac));
+}
+
+static void test_hashes_hmac_sha256_hash_PRF6(void)
+{
+    /* Test Case PRF-6: */
+    const char strPRF6[] = "This is a test using a larger than block-size key and a "
```

Comment:
@BytesGalore try making this and all other strings `static const` to see if it has any effect on your issues on samr21-xpro